pkgs/applications/virtualization/xen/generic/default.nix

```diff
@@ -701,11 +705,14 @@ stdenv.mkDerivation (finalAttrs: {
       # Development headers in $dev/include.
       mit
     ];
-    maintainers = with lib.maintainers; [ sigmasquadron ];
+    # This automatically removes maintainers from EOL versions of Xen, so we aren't bothered about versions we don't explictly support.
```

Comment:
4.16 will be EOL in December. Once that happens, the number will be updated to only support 4.17 and above, but the sources for 4.16 will be kept. Once 4.17 becomes EOL, the files relating to 4.16 will be removed, and so on.

One thing I haven't figured out yet is how to let people know the files were removed from Nixpkgs if they try to install a very EOL version. Should I add an entry to every removed Xen version to top-level/aliases.nix, or would entries in packages.nix in the Xen tree be more appropriate?

Reply:
I guess it depends on what package set people are most likely to get their Xen from. `throw`s in package sets can be awkward because of Hydra eval stuff. I think that it is fine that it turns from a big “this is EOL” insecurity warning to just an error without any special `throw` message, though. We often don’t add `throw`s for packages in package sets.

Reply:
Ah, then we don't need any special setup.